war/src/main/webapp/scripts/hudson-behavior.js

```diff
@@ -372,7 +372,7 @@ function findFollowingTR(node, className, nodeClass) {
         // Supports plugins with custom variants of <f:entry> that call
         // findFollowingTR(element, 'validation-error-area') and haven't migrated
         // to use querySelector
-        if (className === 'validation-error-area' || className === 'help-area') {
+        if (arguments.length <= 2 && (className === 'validation-error-area' || className === 'help-area')) {
```

Comment:
Can you please explain why this works?

Reply:
The reason I ask is that's adding a conditional to run to something that's already a hack meant for backwards compatibility. It may break other stuff.

Reply:
The "arguments" is the native way to get the parameters passed to that method, and the .length will provide information about the number of argument. The original purpose of the if statement (that was added in the table-to-div effort) is to keep the legacy behavior, for caller like "findFollowingTR(element, 'validation-error-area')" like explained in the comment.

The problem is for the "helpButtonOnClick", there are three arguments, but the second one is "help-area". Due to that, it falls into the "legacy" behavior. Instead of just looking for sibling, it will look for sibling and looking "deep" into them. It's working for most of the form components, except (at least) the dropdownList, due to the body being another TR, which is deeply search, but internally there are other form components, leading to the help area being not the correct one. Instead of being the help area from the dropdownList, it's the help area of the first form component from inside the dropdownListBody.

Reply:
> It may break other stuff.

That was the reason I added a warning in the description of this PR:
:warning: Not tested with other component, just with the dropdownList.

It's a regression after the first step for table-to-div, I just provided this fix as I don't know all the other cases that were covered.

Reply:
So IIURC wouldn't it make more sense to exclude based on the value of the third argument (`setting-sibling` for example) , or even only apply the exclusion for `.help-area`, not `.validation-error-area`?

Reply:
Perhaps, I am lacking the background reason to have kind of hack there.

@timja your expertise on this topic seems required ;)

Reply:
Félix wrote this as far as I know, I probably just extracted it out from the mega PR.

Pleading ignorance.

It will be related to the elements moving in the f:entry on tables to div.

Reply:
I'll try to explain the best I can. 
When changing the f:entry widget we moved the `.validation-error-area` and `.help-area` containers. There were many plugins that broke because they queried these containers directly using `findFollowingTR`. The better example was the git-parameters-plugin, where the branch selection box would break (see screenshot to see it working)

<img width="610" alt="Captura de pantalla 2020-10-13 a las 11 31 09" src="https://user-images.githubusercontent.com/5738588/95843526-1705e200-0d48-11eb-8b99-5925c9b820ff.png">

I ran the following searches:
- https://github.com/search?q=org%3Ajenkinsci+-path%3Atarget+-repo%3Ajenkinsci%2Fjenkins+help-area
- https://github.com/search?q=org%3Ajenkinsci+-path%3Atarget+-repo%3Ajenkinsci%2Fjenkins+validation-error-area
- https://github.com/search?q=org%3Ajenkinsci+-path%3Atarget+-repo%3Ajenkinsci%2Fjenkins+findFollowingTr

and based on the output I worked on that fix. 

Hope this makes sense for context and to make an idea of the impact of this.